# Løysingsforslag eksamen S1 - H23 

Her er eit forslag til løysing av hausteksamen i S1 2023. I del 2 er oppgåvene stort sett Python brukt som hjelpemiddel.

Eg kan ikkje lova at løysinga er feilfri... 😊 Gje meg gjerne ein lyd i kommentarfeltet eller [her](/about.qmd) om du ser feil :mag: 

Sist oppdatert: 15.11.2023

---

Miniatyrbilete: 
<a href="https://www.flaticon.com/free-icons/exam" title="exam icons">Exam icons created by Freepik - Flaticon</a>

Eksamenssettet finn du hos UDIR: [Finn eksamensoppgaver](https://sokeresultat.udir.no/eksamensoppgaver.html)


# Del 1 - utan hjelpemiddel

## Oppgåve 1

Skriv så enkelt som mogleg

$$ \begin{align*} 
\left( \frac{3a^2}{2b^3}\right)^2 \cdot \left( \frac{a^2 b^{-5}}{4}\right)^{-1} &= \left( \frac{3^2 a^4}{2^2 b^6}\right) \cdot \left(\frac{a^{-2} b^{5}}{{(2^2)}^{-1}}\right) \\
&= \left(\frac{3^2 a^4 a^{-2} b^5}{2^2 b^6 2^{-2}}\right) \\
&= \frac{3^2 a^2 b^5}{b^6} \\
&= \frac{9a^2}{b}
\end{align*}$$


## Oppgåve 2

Skriv uttrykka i stigande rekkefølge: $$2\ln e^{3} \quad , \quad  3\lg 70 \quad , \quad e^{3\ln 2} $$

Ser på kvart uttrykk for seg.

**Først $2\ln e^3$**
$$2 \ln e^3 = 2\cdot 3 \ln e = 6 \cdot 1 = 6$$

**Vidare $3 \lg 70$**
$$3 \lg 70 = 3 \lg (7\cdot 10) = 3 (\lg 7 + \lg 10) = 3 \lg 7 + 3 $$
Veit at $0<\lg 7 < 1$ så $3 \lg 7 < 3$ og $3 \lg 70 < 6$

**Til slutt $e^{3\ln 2}$**
$$e^{3\ln 2} = {\left(e^{\ln 2}\right)^3} = 2^3 = 8$$

**I stigande rekkefølge:**
 $$3\lg 70 \quad , \quad  2\ln e^{3} \quad , \quad e^{3\ln 2} $$


## Oppgåve 3

Du kastar tre terningar

### a) alle ulike

Terninane har seks sider. Finn sannsynet for at alle terningane viser forskjellige tal på auge. 
$$P(\text{alle ulike}) = \frac{6}{6} \cdot \frac{5}{6} \cdot \frac{4}{6} = 1\cdot \frac{20}{36} = \frac{5}{9}$$  

Eventuelt
$$P(\text{alle ulike}) = \frac{\text{gunstige utfall}}{\text{moglege utfall}} = \frac{6\cdot 5 \cdot 4}{6\cdot 6\cdot 6} = \frac{5}{9}$$

### b) nøyaktig to terningar er like

Det at nøyaktig to terningar er like skjer viss ikkje alle terningane viser ulikt, eller alle terningane viser likt.

$$P(\text{to like}) = 1 - P(\text{alle like}) - P(\text{alle ulike})$$

Må finna $P(\text{alle like})$

$$P(\text{alle like}) = \frac{\text{gunstige utfall}}{\text{moglege utfall}} = \frac{6\cdot 1 \cdot 1}{6\cdot 6\cdot 6} = \frac{1}{36}$$

Dermed kan me finna sannsynet for at nøyaktig to er like:

$$ \begin{align*} 
P(\text{to like}) &= 1 - P(\text{alle like}) - P(\text{alle ulike}) \\
&= 1 - \frac{1}{36} - \frac{5}{9} \\
&= \frac{36}{36} - \frac{1}{36} - \frac{5\cdot 4}{9 \cdot 4} \\
&= \frac{36-1-20}{36} \\
&= \frac{15}{36} \\&= \frac{5}{12}
\end{align*}$$

## Oppgåve 4

Ein funksjon $f$ er gitt ved

$$f(x) = 
\begin{cases} 
x^2+3x-a^2 & \text{for } x < 1 \\
x-1 & \text{for } x \geq 1
\end{cases} $$

Bestem $a$ slik at funksjonen blir kontinuerleg. 

Viss funksjonen skal vera kontinuerleg må 
$$\begin{align*}
\lim_{x\rightarrow 1^{-}}f(x) &= \lim_{x\rightarrow 1^{+}}f(x) \\
1^2 + 3\cdot 1 - a^2 &= 1-1 \\
4-a^2 &= 0 \\
a^2 &= 4 \\
a = \pm 2
\end{align*}
$$

$f(x)$ er kontinuerleg når $a=-2 \vee a=2$


## Oppgåve 5

Ei bedrift produserer ei vare. Dei daglege kostnadane $K$ i kroner ved produksjon av $x$ einingar av vara er gitt ved
$$K(x) = 0,1x^2 + 100x + 9000$$

Den økonomiansvarlege i bedrifta har laga programmet nedanfor. 

In [1]:
#| eval: false
#| output: false
#| code-line-numbers: true
def K(x):
    return 0.1*x**2 + 100*x + 9000

grense = 200
h = 0.00001
a = 1

while (K(a + h) - K(a))/h < grense:
    a = a + 1

print(a)

500


Me kjenner att `K(a + h) - K(a))/h` som definisjonen til den deriverte av $K$. Programmet finn dermed ut kva tid den deriverte til $K$, *grensekostnaden (kompetansemål i S2)*, blir større enn/passerer `grense = 200`. 

Resultatet når ein køyrer programmet vert då den siste x-verdien (heiltal) som gjer $K'(x) \leq 200$. 

Løyser likninga $K'(x)=200$: 

$$\begin{align*} 
K'(x) &= 200 \\
0,2x + 100 &= 200 \\
0,2x &= 200 - 100 \\
x &= \frac{100}{0,2} = 500
\end{align*} $$

Dermed er $K'(500) = 200$ som vil seie at programmet skriv ut `500` når det køyrer, sidan $K'(501)$ vil vera større enn vilkåret i `while`-løkka i rad 8. (sidan $K'(a)$ er ein lineær funksjon med positivt stigningstal). 

Dette fortel bedrifta at om ein aukar produksjonen frå 500 til 501 einingar per veke så vil kostnadane auka med omlag 200 kr. Det vert 200 kr dyrare å laga ei til eining av vara. (og denne grensekostnaden vert større og større sidan $K'(x)$ har positivt stigningstal.  

# Del 2

## Oppgåve 1


Ein møbelfabrikk produserer ein type sofaar. Tabellen viser samanhengen mellom tal sofaar produsert per månad og produksjonskostnadane per månad. 

| Tal sofaar | Produksjonskostnad (i tusen kr) |
|---|---|
|10 |270|
|25 |550|
|40 |870|
|70 |1500|
|140 |2200|
|100 |3300|
| 180|4500|

Fabrikken sel alle sofaane til ei møbelkjede. Dei får 28 000 kr per sofa. 

### a) Finne O(x)

Bruker regresjon til å finna eit uttrykk for $K(x)$ som passar med tala i tabellen. $x$ er talet på sofaar produsert. 


In [1]:
import numpy as np 

# definerer variablar for x og y
x = np.array([10, 25, 40, 70, 100, 140, 180])
y = np.array([270, 550, 870, 1500, 2200, 3300, 4500])

# finn koeffesientane med regresjon (andregradsfunksjon)
a, b, c = np.polyfit(x, y, 2)

print(f"K(x) = {a:.3f}x^2 + {b:.0f}x + {c:.0f}")

K(x) = 0.041x^2 + 17x + 103


Veit at dei får 28000 kr per sofa. Dermed er $$I(x) = 28x$$

Overskotsfunksjonen finn me når me trekk kostnadane frå inntektene. 

$$O(x) = I(x) - K(x) = 28x - (0.041x^2 + 17x + 103) = -0.041x^2 + 11x - 103$$

Modellen i oppgåva er den samme som me kom fram til her, og er dermed ein god modell for overskotet. 

### b) Størst overskot

Finn ekstremalpunktet til $O(x)$ for å finna produksjonsmengda som gjev størst overskot. Sidan andregradsleddet er negativ er grafen konkav og ekstremalpunktet er difor eit toppunkt (*størst overskot*). 

In [2]:
from sympy import Symbol, diff, Eq, Reals, solveset

# definerer x
x = Symbol("x")

O = -0.041*x**2 +11*x -103

# definerer vs og hs av likninga
vs = diff(O)    #O'(x)
hs = 0 

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, x, domain=Reals)

# skriv ut svaret
print(f"x = {løysing.evalf(4)}")

x = {134.1}


Bedrifta får størst overskot om dei produserer 134 sofaar i månaden. 

### c) Lågast salspris

Fabrikken ønskjer at overskotet skal vere 1 000 000 kr per månad. Dei vil endre salsprisen på sofaane. Bestem den lågaste salsprisen dei kan setje per sofa dersom dei skal få dette overskotet. 

Løyser likninga med CAS i Python (`SymPy`).


In [3]:
from sympy import Symbol, diff, Eq, Reals, solveset

# definerer symboler
x = Symbol("x")
a = Symbol("a")

# definerer K(x) og I(x). Ny pris = a. 
K = 0.041*x**2 + 17*x + 103
I = a*x

# Bestemmer O(x)
O = I - K

# Finn likning for ekstremalpunktet til O
vs = diff(O, x)
hs = 0

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, x, domain=Reals)
løysing

Intersection({12.1951219512195*a - 207.317073170732}, Reals)

Ser at $O(x)$ har ekstremalpunkt i $x=12.2a - 207.3$. 

In [4]:
# definerer x-verdien til ekstremalpunktet uttrykt ved a
x_topp = 12.2*a - 207.3

# finn O(x_topp)
nyO = O.subs(x, x_topp)

# Sett opp likning for O(x_topp)=1000 
vs = nyO
hs = 1000

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, a, domain=Reals)
løysing

{3.55039004456709, 30.449616515434}

Ser at $a = 3.55 \vee a = 30.45$. 
Sjekkar kva $x$ er for dei ulike verdiane.  

In [5]:
a1 = 3.55
a2 = 30.45

x1 = x_topp.subs(a, a1).evalf()
x2 = x_topp.subs(a, a2).evalf()

print(f"Pris på {a1*1000} kr gjev x = {x1:.0f}")
print(f"Pris på {a2*1000} kr gjev x = {x2:.0f}")

Pris på 3550.0 kr gjev x = -164
Pris på 30450.0 kr gjev x = 164


Produksjonsmengda, $x$, må vera positiv. Dermed er den lågaste prisen bedrifta kan ta for å få 1M kr overskot 30450 kr pr. sofa. 

::: {.callout-tip collapse="true"}
## Løyse i GeoGebra sin CAS

Her er løysing av 1c i "vanleg" CAS.

![](cas-1c-del2.png){fig-alt="Bilde av 1c løyst i CAS." fig-align="left" width=60%}
:::

## Oppgåve 2

Undersøkingar viser at 10 % av alle menn og 8 % av alle kvinner er venstrehendte.  På ein skule er det 280 gutar og 220 jenter. 

### a) minst 25 gutar er venstrehendte

Bestem sannsynet for at minst 25 av gutane er venstrehendte. 

### b) P(minst tre gutar venstrehendt) > 0.20

Kor mange gutar må det vere i ein klasse dersom sannsynet for at minst tre
av gutane er venstrehendte, skal vere større enn 20 %?

### c) Nøyaktig tre venstrehendte elevar i klassen

I ein klasse er det 13 gutar og 17 jenter. Bestem sannsynet for at nøyaktig tre av elevane i klassen er venstrehendte

## Oppgåve 3

Per og Kåre set inn like store beløp på kvar sin konto. Per får ei årleg rente på
3,00 %, mens Kåre får ei årleg rente på 6,00 %.




### a) finne sparebeløp
Kva for eit beløp må Per setje inn dersom han skal ha 30 000 kroner på kontoen etter 8 år?



Løyser likninga $x\cdot 1.03^8 = 30000$ i CAS:

In [6]:
from sympy import Symbol, Eq, Reals, solveset

# definerer symboler
x = Symbol("x")

# Finn likning
vs = x * 1.03**8
hs = 30000

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, x, domain=Reals)
løysing

{23682.2770294181}

Per må sette inn 23683 kr for å ha 30 000 kr på kontoen etter 8 år. 



### b) dobbelt sparebeløp

**Påstand**

> Det vil gå nøyaktig dobbelt så lang tid før beløpet Per har på konto, har dobla seg, som det vil gå før beløpet Kåre har på konto, har dobla seg.

Argumenter for at påstanden ikkje er riktig.

Dersom dette stemmer må $1.03^{2x}$ og $1.06^x$ vera det same.
$$1.03^{2x} = (1.03^2)^x = 1.0609^x \neq 1.06^x$$

Påstanden er ikkje riktig.

### c) doble totalt sparebeløp

Kor lang tid vil det går før Per og Kåre til saman har dobbelt så mykje pengar
som dei sette inn på kontoane, dersom den årlege renta er høvesvis 3,00 % og 6,00 %?

Kallar innskotet for $I$. Då får me likninga

$$I\cdot 1.03^x + I\cdot 1.06^x = 4I \Leftrightarrow 1.03^x + 1.06^x = 4$$

Løyser i CAS:

In [20]:
from sympy import Symbol, Eq, nsolve

# definerer symboler
x = Symbol("x")

# Finn likning
vs = 1.03**x + 1.06**x
hs = 4

# set opp likninga, og løyser ho numerisk
likning = Eq(vs, hs)
løysing = nsolve(likning, x, 1)
print(løysing)


15.2433642255997


Viss dei får rente ein gong i året vil beløpet ha (litt meir enn) dobla seg i løpet av 16 år. 

## Oppgåve 4

Du kastar fem terningar.

### a) minst to like

Bestem sannsynet for at minst to av terningane viser same tal på auge.
$$P(\text{minst to like}) = 1 - P(\text{ingen like})$$

Reknar ut med python:

In [2]:
# gunstige / moglege utfall
ingen_like = (6*5*4*3*2) / (6**5)

minst_to_like = 1 - ingen_like

print(f"P(minst to like) = {minst_to_like:.4f}")

P(minst to like) = 0.9074


Sannsynet for minst to like terningar er 0.907. 

### b) sum auge

La X vere summen av talet på auge på dei fem terningane.
Bruk programmering til å bestemme $P(X>20)$.

Bruker simulering til å svara på dette:


In [30]:
import numpy as np

# definerer tilfeldighetsgeneratoren
rng = np.random.default_rng()

# tal simuleringar 
N = 10000000

# triller 5 terningar N gongar
terningar = rng.integers(1, 7, size=(5, N))

# finn summen av kvart forsøk
fem_kast = terningar.sum(axis=0)

# tel opp kor mange gongar summen vart større enn 20
gunstige = sum(fem_kast > 20)

# finn sannsynet og skriv ut
p = gunstige / N

print(f"P(X > 20) = {p:.4f}")

P(X > 20) = 0.2217


Finn at $P(X>20)\approx 0.221$. Talet simuleringar er så høgt (10 000 000) at dette sannsynet er ganske nøyaktig.

::: {.callout-tip collapse="true"}
## Forklaring av kodelinjer 10-13 ⬆️

Her bruker eg ei todimensjonal array/tabell for å rekna ut sannsynet. 

Trillar terningar: 

In [38]:
N = 10

terningar = rng.integers(1, 7, size=(5, N))

print(terningar)

[[4 1 2 5 1 4 2 3 6 3]
 [6 1 2 2 4 4 1 4 5 2]
 [2 4 5 2 4 3 3 1 4 2]
 [4 2 5 4 1 6 3 3 2 6]
 [5 6 2 4 4 6 5 2 3 4]]


Ser at denne tabellen får 5 rader og `N` kolonner. 

Finn vidare summen av kvar kolonne:

In [39]:
kolonnesum = terningar.sum(axis=0)
print(kolonnesum)

[21 14 16 17 14 23 14 13 20 17]


:::



### c) bestemme k

Bestem den største verdien av k som er slik at $P(X\geq k )> 0,8$ 

Bruker simuleringar og ser på ulike verdiar for k. Gjer koden frå (b) om til ein funksjon som returnerer sannsynet for at P(X>= k). 

In [40]:
3 >= 1

True

## Oppgåve 5

## Oppgåve 6

Samla løn Per kan me finna på ulike måtar.

Dersom me går ut frå at Per føl tabellen nøyaktig, dvs. 272,55 kr pr. time i 2008 og 2009, 285,50 kr pr. time i 2010, 2011, 2012 osv. vert den samla løna hans:

In [5]:
def periodeløn(tal_år, timeløn):
    return tal_år * 1700 * timeløn

# løn kvar periode
p1 = periodeløn(2010-2008, 272.55)
p2 = periodeløn(2013-2010, 285.50)
p3 = periodeløn(2015-2013, 307.30)
p4 = periodeløn(2019-2015, 314.00)
p5 = periodeløn(2022-2019, 327.60)
p6 = periodeløn(2023-2022, 340.10)

samla_løn = p1 + p2 + p3 + p4 + p5 + p6

print(f"Samla løn, Per: {samla_løn:.2f}")

Samla løn, Per: 7811670.00


Om me på den andre sida bruker modellen for timeløn som me fann i **b)** så kan me bruka den:

In [6]:
samla_løn = 0

# modell for timeløn
def g(x):
    return 278.5 * 1.0151**x

# i frå og med 0 til og med 14
# finn ny årsløn og legg til samla løn dei 15 åra
for i in range(15):
    samla_løn += 1700 * g(i)

print(f"Samla løn, Per: {samla_løn:.2f}")

Samla løn, Per: 7903823.68


Ser at skilnaden på dei to strategiane er rundt 92 000 kr i løpet av dei 15 åra (f.o.m. 2008, t.o.m. 2022). 

## Oppgåve 2

### a) Påstand 1 - logaritmar

La $x=e$. Då er $x>0$ og $\ln(x)=\ln(e)=1$. Dermed får me

$$
\begin{align*}
    (\ln(x))^4 &= 4\ln(x) \\
    1^4 &\neq 4\cdot 1
\end{align*}
$$

**Påstanden er usann**

### b) Påstand 2 - fjerdegradsfunksjonar

Talet på ekstremalpunkt til ein funksjon er knytt saman med talet på nullpunkt til den deriverte til funksjonen. 

Den deriverte til ein fjerdegradsfunksjon er ein tredjegradsfunksjon. Alle tredjegradsfunksjonar har minst eitt nullpunkt. 

Sidan alle tredjegradsfunksjonar har minst eitt nullpunkt har alle fjerdegradsfunksjonar minst eitt ekstremalpunkt. 

**Påstanden er sann**

### c) Påstand 3 - Lotto

Kan sjå på dette som hypergeometrisk sannsyn, der populasjon og utval er delt i to grupper, *mindre enn 18* og *ikkje mindre enn 18*.

Det er 17 av tala som er *mindre enn 18*, $0\rightarrow 17$, og 17 tala som er *ikkje mindre enn 18*, $18\rightarrow 34$. 

La $X$ vera antall kuler mindre enn 18. Då er 

$$P(X=7) = \frac{\left(17 \atop 7 \right) \left(17 \atop 0 \right)}{\left(34 \atop 7\right)}$$

og samtidig er 

$$P(X=0) = \frac{\left(17 \atop 0 \right) \left(17 \atop 7 \right)}{\left(34 \atop 7\right)}$$

Dermed ser me at $P(X=7) = P(X=0)$. 

**Påstanden er sann**

## Oppgåve 3

Eit spel med to terningar. Mona finn sannsyn med programmet




In [8]:
#| code-line-numbers: true
g = 0
for i in range(1, 7):
    for j in range(1, 7):
        if i+j >= 8:
            g = g + 1

print(g/36)

0.4166666666666667


### a) Sannsyn i programkoden

Med dette programmet tel Mona opp kor mange av dei moglege utfalla (summen av to terningar) som blir 8 eller større. 
Løkka går gjennom alle kombinasjonar av dei to terningane `i=1, j=1`, `i=1, j=2`, ..., `i=2, j=1` osb.


### b) Nytt program - tre terningar

Eit anna spel med tre terningar. Gevinst om summen av auge er 7 eller 11. 
Kan utvida programmet frå **a)**. Endrar nemnaren til 216 ($6\cdot 6 \cdot 6$) i siste linja. 

In [9]:
#| code-line-numbers: true
g = 0
for i in range(1, 7):
    for j in range(1, 7):
        for k in range(1, 7):
            if (i+j+k == 7) or (i+j+k == 11):
                g = g + 1

print(g/216)

0.19444444444444445


Sannsynet for å vinna dette spelet er $0,194$. 

## Oppgåve 4

### a) Pris 40 kr

Viss $p=40$ les me av grafen at $e=2100$. 

Det vil seie at ein sel 2100 einingar til 40 kr pr. stk. 

Daglege inntekter på $2100 \cdot 40 kr = 84 000 kr$. 

### b) Inntekt 75 000 kr

Ser av grafen at $e(p)$ har konstantledd 4500 og stigningstal -60. (Nedgang frå 4500 til 3900 på y-aksen når ein går frå 0 til 10 på x-aksen)

Dvs. $e(p) = 4500 - 60p$. 

Daglege inntekter er tal eiingar gange prisen, $e(p)\cdot p$.

Løyser likninga $e(p)\cdot p = 75000$.



In [10]:
from sympy import Symbol, diff, Eq, Reals, solveset

# definerer p
p = Symbol("p")

# definerer vs og hs av likninga
vs = (4500 - 60*p)*p
hs = 75000

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, p, domain=Reals)

# skriv ut svaret
print(f"p = {løysing.evalf(4)}")

p = {25.0, 50.0}


Ser at om prisen er 25 kr eller 50 kr så vil dei daglege inntektene bli 75000 kr. 

### c) Størst inntekt 

Sidan funksjonen for daglege inntekter er ein konkav andregradsfunksjon ($(4500-60p)p = -60p^2+45000$) er ekstremalpunktet eit toppunkt. Av symmetri i andregradsfunksjonar følgjer det at toppunktet ligg midt mellom 25 kr og 50 kr. 

Pris på $(25 kr + 50 kr)/2 = 37,5 kr $ gjev størst inntekt. 

## Oppgåve 5

### a) Bestem sannsyn

Her kan me gå ut frå at det binomisk sannsyn sidan me har
- to utfall i kvart delforsøk, kvar bilett (brukt/ikkje brukt)
- likt sannsyn for at billetten vert brukt (0,45) i kvart delforsøk
- uavhengige delforsøk (viss me går ut frå at dei som får gratisbillettane ikkje påverkar kvarande med om dei bruker eller ikkje bruker billetten...)

Lar $X$ vera tal billettar som vert brukt. 

Finn $P(X\geq 600)$ ved simulering


In [11]:
import numpy as np
rng = np.random.default_rng()

N = 10000000 # tal simuleringar
p = 0.45     # P(brukt billett)
n = 1300     # tal gratisbillettar

# simulerer 
brukte_billettar = rng.binomial(n, p, size=N)

# finn sannsynet
gunstige = sum(brukte_billettar >= 600)
sannsyn = gunstige / N
print(f"P(X ≥ 600) = {sannsyn:.4f}")

P(X ≥ 600) = 0.2097


Sannynet for at minst 600 av billettane vert brukt er omlag 0,21.

### b) Finn tal gratisbilettar

Vil finna ut kor mange gratisbillettar dei må dela ut dersom sannsynet for at minst 600 blir brukt skal vera over 95 %. 

Gjer simuleringa frå **a)** til ein funksjon og testar med fleire og fleire gratisbillettar til sannsynet for at minst 600 blir brukt passerer 95 %. 

In [12]:
# sannsynsfunksjon
def sannsyn(n):
    N = 1000000
    p = 0.45

    # simulerer 
    brukte_billettar = rng.binomial(n, p, size=N)

    # finn sannsynet
    gunstige = sum(brukte_billettar >= 600)
    sannsyn = gunstige / N

    return sannsyn

n = 1300

# aukar tal gratisbillettar til sannsynet er 95% eller meir
while sannsyn(n) < 0.95:
    n += 1

print(n)

1401


Dvs. om dei deler ut 1400 bilettar vil det vera over 95 % sannsynleg at minst 600 av dei vert brukt. 

## Oppgåve 6

Samanhengen mellom lydstyrken $L$ (målt i $dB$) og lydintensiteten $I$ (målt i $W/m^2$) er gitt ved

$$L = 120 + 10\cdot \lg(I)$$

Smertegrense for menneske på kring 130 $dB$

### a) Bestem lydintensitet

Set inn $L=130$ og løyser likninga for $I$. 

In [13]:
from sympy import Symbol, diff, Eq, Reals, solveset, log

# definerer I
I = Symbol("I")

# definerer vs og hs av likninga
vs = 130
hs = 120 + 10*log(I, 10)

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, I, domain=Reals)

# skriv ut svaret
print(f"I = {løysing.evalf(4)}")

I = {10.0}


Lydintensiteten er $10 W/m^2$ når lydstyrken er $130dB$.

### b) Auke i intensitet

Finn intensiten ved $132dB$. 

In [14]:
# definerer vs og hs av likninga
vs = 132
hs = 120 + 10*log(I, 10)

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, I, domain=Reals)

# skriv ut svaret
print(f"I = {løysing.evalf(4)}")

I = {15.85}


Lydinsiteten har auka frå 10 til 15,85. Det gir oss vekstfaktor $15,85 / 10 = 1,581$. 

Auken i lydintensitet er omlag 58,1 % når ein aukar lydstyrken med $2 dB$. 

### c) Minste avstand til flyet

Effekten til lyd frå lydkjelde, $E$, vil gje lydintensitet $I$ i ein avstand $r$ (målt i $m$) frå lydkjelda vera $$I = \frac{E}{4\pi \cdot r^2}$$. 

Lydstyrke frå eit fly er $140 dB$ når ein er $50 m$ frå flyet. 

Bestem den minste avstanden til flyet der lydstyrken er mindre enn $130dB$. 

Finn intensiteten ved $140 dB$: 



In [15]:
# definerer vs og hs av likninga
vs = 140
hs = 120 + 10*log(I, 10)

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, I, domain=Reals)

# skriv ut svaret
print(f"I = {løysing.evalf(4)}")

I = {100.0}


Vidare bruker me at $I=100$ for å finna E. 
$$\begin{align*}
I &= \frac{E}{4\pi\cdot r^2} \\
100 &= \frac{E}{4\pi\cdot 50^2} \\
E &= 100 \cdot 4\pi\cdot 50^2
\end{align*} $$

Dette gjev oss likninga 

$$\begin{align*}
L &= 120 + 10\lg(I) \\
&= 120 + 10\lg\left(\frac{E}{4\pi\cdot r^2}\right) \\
&= 120 + 10\lg\left(\frac{100 \cdot 4\pi\cdot 50^2}{4\pi\cdot r^2}\right) \\
&= 120 + 10\lg\left(\frac{100 \cdot50^2}{r^2}\right)
\end{align*} $$

Løyser likninga for $r$ når $L=130$. 

In [16]:
# definerer r
r = Symbol("r")

# definerer vs og hs av likninga
vs = 130
hs = 120 + 10*log((100*50**2)/(r**2), 10)

# set opp likninga, og løyser ho
likning = Eq(vs, hs)
løysing = solveset(likning, r, domain=Reals)

# skriv ut svaret
print(f"r = {løysing.evalf(4)}")

r = {-158.1, 158.1}


Den minste avstanden er 158 meter. 